RockPaperScissors/RockPaperScissors/RockScissorsPaperGameManager.swift

```diff
@@ -0,0 +1,127 @@
+//
+//  RockScissorsPaperGameManager.swift
+//  RockPaperScissors
+//
+//  Created by Harry, Andrew on 2022/12/28.
+//
+import Foundation
+
+class RockScissorsPaperGameManager {
+    func printMenu() {
+        print("가위(1), 바위(2), 보(3)! <종료: 0>", terminator: ": ")
+    }
+    
+    func readUserInput() -> String {
+        guard let userInput = readLine() else {
+            return ""
+        }
+        
+        return userInput
+    }
+    
+    func validateUserInput(userInput: String) throws -> Int {
+        guard userInput != "",
+              let number = Int(userInput),
+              number >= 0, number <= 3 else {
+            throw InputError.invalidInput
+        }
+        
+        return number
+    }
+    
+    func generateComputerHand() -> HandShape? {
+        guard let computerHand = HandShape.allCases.randomElement() else {
+            return nil
+        }
+        
+        return computerHand
+    }
+    
+    func generateUserHand(validationResult: Int) -> HandShape? {
+        let handShapeList: [HandShape?] = HandShape.allCases
+        
+        guard let userHand = handShapeList[validationResult - 1] else {
+            return nil
+        }
+        
+        return userHand
+    }
+    
+    func compareHandShape(computerHand: HandShape?, userHand: HandShape?) -> MatchResult? {
+        guard let computerHand = computerHand, let userHand = userHand else {
+            return nil
+        }
+        
+        switch (computerHand, userHand) {
+        case (.rock, .scissors),
+            (.scissors, .paper),
+            (.paper, .rock):
+            return .lose
+        case (.scissors, .rock),
+            (.paper, .scissors),
+            (.rock, .paper):
+            return .win
+        default:
+            return .same
+        }
+    }
+    
+    func informMatchResult(matchResult: MatchResult?) -> String? {
+        if let matchResult = matchResult {
+            switch matchResult {
+            case .win:
+                print("이겼습니다!")
+                return "사용자"
+            case .same:
+                print("비겼습니다!")
+                return nil
+            case .lose:
+                print("졌습니다!")
+                return "컴퓨터"
+            }
+        }
+        return nil
+    }
+    
+    func playRockPaperScissors() -> (matchResult: MatchResult?, gameFlow: GameFlow) {
+        printMenu()
+        let input = readUserInput()
+        var userNumber = 0
+        do {
+            userNumber = try validateUserInput(userInput: input)
+            if userNumber == 0 {
+                print("게임 종료")
+                return (nil, .gameOver)
+            }
```

Comment:
```suggestion
        let input = readUserInput()
        var userNumber = 0
        
        do {
            userNumber = try validateUserInput(userInput: input)
            if userNumber == 0 {
                print("게임 종료")
                return (nil, .gameOver)
            }
```
변수와 메서드 부분은 이렇게 구분하는 게 좋아보입니다!

Reply:
개행 하였습니다!